Tests/MongoSwiftTests/MongoConnectionStringTests.swift

```diff
@@ -661,4 +661,42 @@ final class ConnectionStringTests: MongoSwiftTestCase {
         connString7.credential?.source = nil
         expect(connString7.description).toNot(contain("authsource"))
     }
+
```

Comment:
for the connection string only case these already have good coverage from the spec tests, but I figured I would add some additional testing here around setting the options programmatically.